inst/include/dust/dust.hpp

```diff
@@ -141,8 +181,66 @@ class filter_state {
   std::vector<real_t> history_value;
   std::vector<size_t> history_order;
 };
+
+template <typename real_t>
+struct device_state {
+  void initialise(size_t n_particles, size_t n_state) {
+    const size_t n_rng = dust::rng_state_t<real_t>::size();
+    // NOTE: not setting up yi_selected here, which was used in dustgpu
+    y = dust::device_array<real_t>(n_state * n_particles);
+    rng = dust::device_array<uint64_t>(n_rng * n_particles);
+  }
+  dust::device_array<real_t> y;
+  dust::device_array<uint64_t> rng;
+};
+
+// We need to compute the size of space required for integers and
+// reals on the device, per particle. Because we work on the
+// requirement that every particle has the same dimension we pass an
+// arbitrary set of shared parameters (really the first) to
+// device_work_size. The underlying model can overload this template
+// for either real or int types and return the length of data
+// required.
+template <typename T>
+size_t device_work_size_int(typename dust::shared_ptr<T> shared) {
```

Comment:
Can we clarify the name here that this is device_shared_size? Not the internal size

Reply:
This is the internal size